mod-source-record-storage-server/src/main/java/org/folio/dao/RecordDaoImpl.java

```diff
@@ -763,6 +767,34 @@ private void appendWhere(SelectJoinStep step, ParseLeaderResult parseLeaderResul
         .onFailure(v -> txQE.rollback());
   }
 
+  @Override
+  public Future<MarcBibCollection> verifyMarcBibRecords(List<String> marcBibIds, String tenantId) {
+    if (marcBibIds.isEmpty()) {
+      return Future.succeededFuture(new MarcBibCollection());
+    }
+    var marcHrid = DSL.field("marc.hrid");
+
+    return getQueryExecutor(tenantId).transaction(txQE -> txQE.query(dsl ->
+      dsl.selectDistinct(marcHrid)
+        .from("(SELECT unnest(" + DSL.array(marcBibIds.toArray()) + ") as hrid) as marc")
+        .leftJoin(RECORDS_LB)
+        .on(RECORDS_LB.INSTANCE_HRID.eq(marcHrid.cast(String.class)))
+        .where(RECORDS_LB.INSTANCE_HRID.isNull())
```

Comment:
Could you please modify the query by including condition of `recordType == MARC_BIB`